test_suite/package/otel_package/fixtures.py

```diff
@@ -0,0 +1,111 @@
+import json
+import time
+import pytest
+from lib import console_lib
+from lib import test_lib
+from test_suite.generic import helpers
+
+
+@pytest.fixture(scope='class')
+def initialize_veribales(request, host):
```

Comment:
"you were having issue when using fixtures inside another fixture." - This is exactlly the issue I replied on your comment above (pytest doen't support using fixtures of function scope inside a class scope fixture for example) this is why I added this as a function rather using this "instance_data" fixture.

In my opinion this fixture is efficient in the way that you declare those variables once in the code run for Otel, this way they become the class variables and can be called by self instead of getting them each time in every function/fixture or test in the same code run.
If you notice I need to use these variables in different fixtures/functions many times.

If you have a better working suggestion to solve this issue let's talk about it cause I've tried not a few (by using the existing "instance_data") and this was the most efficient so far that could work well.

Reply:
Yes, it makes sense. Thanks for the explanation. You're right, the code is more efficient this way.
The only thing is that `instance_data` fixture is used _everywhere_ on purpose, for metadata generation in the reports and it should be included on each test case, because this metadata is needed for analysis purposes.
One example is in the instance JSON metadata that can be seen here:
![image](https://github.com/user-attachments/assets/e2e911ef-a838-4349-8dc9-8efa8a6fc781)

This metadata is also used by our analyzers to know about the context on which the test was run, and extract the data needed (e.g. the AMI ID, region, etc.).
The above are the real reasons why we use scope function and `autouse=True`. So, given that it's already there, I thought you can just use it instead of declaring the variables, which would be redundant.

Also I don't know the class variables approach would behave in the case we run tests in parallel on multiple instances. Although it is not the case for packages testing (yet?).

Reply:
If we change the scope of the fixture to 'class', it will be executed once per test class instead of once per test function.